src/components/translatable/ReferenceFilters.js

```diff
@@ -0,0 +1,113 @@
+  const { refSelected, cvSelected } = filterList[index].reduce(
+    (values, filter) => {
+      const [type, value] = filter.split("|")
+      if (type === "ref") {
+        values.refSelected.push(value);
+      }
+      if (type === "cv") {
+        values.cvSelected.push(value);
+      }
+      return values;
+    },
+    { cvSelected: [], refSelected: [] }
+  );
+
+  const [c, v] = cvSelected?.[0]?.split(":") || [];
+
+  const [references, setReferences] = useState(refSelected || []);
+  const [chapter, setChapter] = useState(c || "All");
+  const [verses, setVerses] = useState(v || "All");
+
+  const buildCv = (c, v) => {
+    if (c === "All") return [];
+    const ref = v === "All" ? `${c}` : `${c}:${v}`;
+    return [`cv|${ref}`];
+  };
+
+  const handleChange = (cv, references) => {
+    const ref = references.map((value) => (`ref|${value}`));
+    filterList[index] = cv.length || ref.length ? [...cv, ...ref] : [];
+    onChange(filterList[index], index, column);
+  };
+
+  const handleChangeReferences = (event, newValue) => {
+    setReferences(newValue);
+    const cv = buildCv(chapter, verses);
+    handleChange(cv, newValue);
+  };
+
+  const handleChangeChapter = (event, newValue) => {
+    const newChapter = newValue || "All";
+    setChapter(newChapter);
+    setVerses("All");
+    const cv = buildCv(newChapter, verses);
+    handleChange(cv, references);
+  };
+
+  const handleChangeVerse = (event, newValue) => {
+    const newVerses = newValue || "All";
+    setVerses(newVerses);
+    const cv = buildCv(chapter, newVerses);
+    handleChange(cv, references);
+  };
+
+  return (
+    <>
+      <FormControl>
+        <Autocomplete
+          onChange={handleChangeReferences}
+          id="reference-filter"
+          value={references || []}
+          freeSolo
+          multiple
+          options={values.raw}
+          renderInput={(params) => (
+            <TextField {...params} label={column.label} />
+          )}
+        />
+      </FormControl>
+      <FormControl>
+        <Box
+          sx={{
+            mt:"1em",
+            display: "grid",
+            gridTemplateColumns: "1fr 1fr",
+            gap: "58px"
+          }}
+        >
+          <Autocomplete
+            value={chapter}
+            style={{ overflow: "visible" }}
+            options={["All", ...Object.keys(values.cv)]}
+            getOptionLabel={(option) => `${option}`}
+            onChange={handleChangeChapter}
+            handleHomeEndKeys
+            renderInput={(params) => (
+              <TextField {...params} label={"Chapter"} />
+            )}
+          />
+          <Autocomplete
+            value={verses}
+            style={{ overflow: "visible" }}
+            options={["All", ...(values.cv[chapter] || [])]}
+            getOptionLabel={(option) => `${option}`}
+            onChange={handleChangeVerse}
+            handleHomeEndKeys
+            renderInput={(params) => <TextField {...params} label={"Verses"} />}
```

Comment:
I think this label should be singular since the field only accept one verse and does not support multiple values or a verse range.
```suggestion
            renderInput={(params) => <TextField {...params} label={"Verse"} />}
```

Reply:
Yes I changed it to plural because it does support range values, but only if there are rows that have a range reference. It takes those values from the existing rows, if there's one row with reference 1-3 it will add that as an option.

To test this, edit the reference field of one note and make it a range, then try to filter it using chapter and verses input fields.

Reply:
Ok, I see. This is G2G then. Good work!